File: backend/src/main/java/com/profroid/profroidapp/employeesubdomain/presentationLayer/employeePresentationLayer/EmployeeResponseModel.java
package com.profroid.profroidapp.employeesubdomain.presentationLayer.employeePresentationLayer;

import com.profroid.profroidapp.employeesubdomain.dataAccessLayer.employeeDataAccessLayer.EmployeeAddress;
import com.profroid.profroidapp.employeesubdomain.dataAccessLayer.employeeDataAccessLayer.EmployeeIdentifier;
import com.profroid.profroidapp.employeesubdomain.dataAccessLayer.employeeDataAccessLayer.EmployeePhoneNumber;
import com.profroid.profroidapp.employeesubdomain.dataAccessLayer.employeeDataAccessLayer.EmployeeRole;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class EmployeeResponseModel {

    private EmployeeIdentifier employeeIdentifier;

    private String firstName;
    private String lastName;
    private String userId;

    private List<EmployeePhoneNumber> phoneNumbers;
    private EmployeeAddress employeeAddress;

    private EmployeeRole employeeRole;
}

File: frontend/src/features/employee/models/EmployeeResponseModel.ts
import type { EmployeeAddress } from "./EmployeeAddress";
import type { EmployeeIdentifier } from "./EmployeeIdentifier";
import type { EmployeePhoneNumber } from "./EmployeePhoneNumber";
import type { EmployeeRole } from "./EmployeeRole";


export interface EmployeeResponseModel {
  employeeIdentifier: EmployeeIdentifier;
  firstName: string;
  lastName: string;
  userId: string;

  phoneNumbers: EmployeePhoneNumber[];
  employeeAddress: EmployeeAddress;
  employeeRole: EmployeeRole;
}
